### PR TITLE
Rake taks to gather information about all objects in our Fedora 3 repo

### PR DESCRIPTION
**Pull request by hectorcorrea**
Rake taks to gather information about all objects in our Fedora 3 repo to audit our data before and after a migration.

Running `bundle exec rake pre_migrate:f3_audit` will dump to a SQL table the list of all objects (in our namespace) in our Fedora 3 repo. The table will include the PID, the model (e.g. GenericFile, Batch), and the title of each object.

This code does not use neither Rubydora nor ActiveRecord. It issues straight HTTP calls to the Fedora API to fetch the information (although I stole from Rubydora's Search implementation how to get the pagination to work.) This code does NOT fetch the content of the objects, it only fetches the XML with their metadata. 

There are a couple of rake tasks that only query the Fedora 3 repo and dump the information to the console if you want just to test the code. `bundle exec rake pre_migrate:get_pids` gives you a list of PIDS whereas `bundle exec rake pre_migrate:get_info` give you a list of PIDs with their model and title.

### CODE REVIEW

**Comment by hectorcorrea** [comment]
Could somebody merge this PR? (Notice that it targets a branch still running on Fedora 3 since it contains Rake tasks to run against the Fedora 3 repo to gather some basic information about the objects that will be migrated to Fedora 4)

**Comment by mjgiarlo** [comment]
@cam156 @awead @mtribone has anyone had a chance to review this yet?  I haven't.  Sorry for the delay, @hectorcorrea

**Comment by hectorcorrea** [comment]
@mjgiarlo no. I thought this was the one that you already had looked at but that was another rake task. But since you are here...could you give it a look and merge it, err, provide some feedback.

File: db/schema.rb [hunk 1]
```diff
@@ -40,10 +40,9 @@
     t.text     "value"
     t.datetime "created_at"
     t.datetime "updated_at"
+    t.string   "external_key"
   end
 
-  add_index "content_blocks", ["name"], name: "index_content_blocks_on_name", unique: true, using: :btree
-
```
Comment (awead) [inline db/schema.rb:46]: Is this related to your tables?

File: db/schema.rb [hunk 2]
```diff
@@ -40,10 +40,9 @@
     t.text     "value"
     t.datetime "created_at"
     t.datetime "updated_at"
+    t.string   "external_key"
```
Comment (awead) [inline db/schema.rb:43]: This also seems unrelated to adding a new table. Maybe Rails is doing something new here?

File: db/schema.rb [hunk 3]
```diff
@@ -214,8 +222,8 @@
     t.string   "label"
     t.string   "lowerLabel"
     t.string   "uri"
-    t.datetime "created_at", default: '2014-12-10 16:14:48'
-    t.datetime "updated_at", default: '2014-12-10 16:14:48'
+    t.datetime "created_at", default: '2014-08-20 15:18:33'
+    t.datetime "updated_at", default: '2014-08-20 15:18:33'
```
Comment (awead) [inline db/schema.rb]: I don't think the date values matter, but we should be sure.

File: app/models/migrate_audit.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class MigrateAudit < ActiveRecord::Base
+
+  def self.f3_audit(fedora_url, fedora_user, fedora_password, namespace)
```
Comment (awead) [inline app/models/migrate_audit.rb:5]: You could get all these values from config files, I believe.

File: app/models/migrate_audit_fedora3.rb [hunk 5]
```diff
@@ -0,0 +1,126 @@
+class MigrateAuditFedora3
+
+  def initialize(fedora_url, fedora_user, fedora_password, namespace)
```
Comment (awead) [inline app/models/migrate_audit_fedora3.rb:3]: Same as above comment. You could get these values from config files.

**Comment by mtribone** [comment]
I'd be more than happy to click the shiny, green button  :-)

File: app/models/migrate_audit_fedora3.rb (hunk 5 above)
Comment (hectorcorrea) [inline app/models/migrate_audit_fedora3.rb:3]: @awead I tried to keep this class (and the Model) as free of dependencies as possible and that's why I am passing these values as parameters. In the rake task that instantiate these classes I am indeed reading these parameters from the config as you suggested.

**Comment by awead** [comment]
@mjgiarlo @cam156 I've made some notes. The approach looks sound to me. There are no tests, however, and while I know @hectorcorrea has been testing it against our production data, it might be a good idea to have a few integration tests that use Rubydora to verify the counts.

File: db/schema.rb (hunk 3 above)
Comment (hectorcorrea) [inline db/schema.rb]: @awead I'll take a closer look into the changes to schema.rb -- all of them were automatically added by Rails when I ran the generator to create the new `migrate_audits` table but the diff makes it look like it changed other tables which I am not sure is correct.
